src/Rings/orderings.jl

```diff
@@ -1312,7 +1317,7 @@ julia> canonical_matrix(o2)
 [0   -1   0]
 ```
 """
-function canonical_matrix(M::MonomialOrdering)
+@attr ZZMatrix function canonical_matrix(M::MonomialOrdering)
   return canonical_matrix(nvars(base_ring(M)), M.o)
 end
```

Comment:
```suggestion
function canonical_matrix(M::MonomialOrdering)
  if !isdefined(M, :canonical_matrix)
    M.canonical_matrix = canonical_matrix(nvars(base_ring(M)), M.o)
  end
  return M.canonical_matrix
end
```

Reply:
In general, if s.th. should be stored in almost all instances, I'd not put it in attributes, but into the structure. This allows for discovery...